### PR TITLE
Add rec and source to capi.

### PR DESCRIPTION
**Pull request by tdaede**

### CODE REVIEW

File: src/capi.rs [hunk 1]
```diff
@@ -218,7 +218,7 @@ impl EncContext {
   fn receive_packet(&mut self) -> Result<Packet, rav1e::EncoderStatus> {
     fn receive_packet<T: rav1e::Pixel>(
       ctx: &mut rav1e::Context<T>,
-    ) -> Result<Packet, rav1e::EncoderStatus> {
+    ) -> Result<Packet, rav1e::EncoderStatus> where FrameInternal: From<Arc<v_frame::frame::Frame<T>>> {
```
Comment (lu-zero) [inline src/capi.rs:221]: Why do you need to express that bound?

Comment (tdaede) [reply src/capi.rs:221]: I am not sure, it is pretty gross. The compiler suggested it to me and it doesn't work without. Is there a way to tell it that the From trait is implemented on all the FrameInternal variants?

**Review by lu-zero — APPROVED** [review]

File: src/capi.rs (hunk 1 above)
Comment (lu-zero) [reply src/capi.rs]: Now I see why, either we do the blanket for every `T: Pixel` or keep this.

**Review by lu-zero — APPROVED** [review]
Looks fine.

**Review by lu-zero — APPROVED** [review]

**Pull request merged**